### PR TITLE
Ensure we can specify different CDI CR name.

### PR DESCRIPTION
**Pull request by awels**
Fix bug where the CR name was hard coded to CDI and it failed in HCO.

Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
HCO names the CDI CR a different name than CDI, and we had hard coded a lookup of the CDI CR as cdi, and it failed when deployed with HCO. This PR fixes that issue, and adds some code so we can specify a CI lane that has a different cr name so we can guard against this happening again.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/util.go [hunk 1]
```diff
@@ -462,11 +463,12 @@ func filterCloneSourcePods(input []v1.Pod) []v1.Pod {
...
-	crKey := client.ObjectKey{Namespace: "", Name: "cdi"}
-	if err := c.Get(context.TODO(), crKey, cr); err != nil {
+	crList := &cdiv1.CDIList{}
+	if err := c.List(context.TODO(), crList, &client.ListOptions{}); err != nil {
 		return nil, err
 	}
-
-	return &cr.Spec.Workloads, nil
+	if len(crList.Items) != 1 {
+		return nil, fmt.Errorf("Number of CDI CRs != 0")
```
Comment (aglitke) [inline pkg/controller/util.go:471]: != 1 should be the message

Comment (awels) [reply pkg/controller/util.go:471]: Yup fixed.

**Review by aglitke — APPROVED** [review]

**Pull request merged**